yolo/yolov8_detection.py

```diff
@@ -189,27 +240,48 @@ def run_yolov8():
 
     st.sidebar.title("Settings")
 
- 
-     # confidence slider
-    confidence = st.sidebar.slider('Confidence', min_value=0.1, max_value=1.0, value=.45)
+    # confidence slider
+    confidence = st.sidebar.slider(
+        'Confidence', min_value=0.1, max_value=1.0, value=.45)
+
+    model = load_model("yolov8s.pt")
 
-    model = YOLO("yolov8n.pt")
     # custom classes
     if st.sidebar.checkbox("Custom Classes"):
         model_names = list(model.names.values())
-        assigned_class = st.sidebar.multiselect("Select Classes", model_names, default=[model_names[0]])
+        assigned_class = st.sidebar.multiselect(
+            "Select Classes", model_names, default=[model_names[0]])
         classes = [model_names.index(name) for name in assigned_class]
     else:
         classes = list(model.names.keys())
 
     st.sidebar.markdown("---")
 
-        # input options
-    input_option = st.sidebar.radio("Select input type: ", ['image', 'video', 'webcam', "YouTube Video"])
+    model_options = ["YOLOv8n", "YOLOv8s", "YOLOv8m", "YOLOv8l", "Custom"]
+
+    yolo_model_selection = st.sidebar.selectbox("Select Yolov8 Model", model_options)
+
+    if yolo_model_selection == "Custom":
+        yolo_model = modules.moduleFileManagement.gatherFilePath('**/best.pt') 
+    else:
+        yolo_model = f"yolov8{yolo_model_selection.lower()[6:]}.pt"
+
+    if not os.path.exists(yolo_model):
+        st.error(f'Go to Offline Data to train your own data set or unzip the "runs" folder')
```

Comment:
including an error message is create, clear indicator whether something is missing!